coinmarketcap/src/adapter.ts

```diff
@@ -0,0 +1,41 @@
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { dominance, marketcap, price } from './endpoint'
+
+const inputParams = {
+  endpoint: false,
+}
+
+export const execute: ExecuteWithConfig = async (request, config) => {
+  const validator = new Validator(request, inputParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+
+  switch (endpoint) {
+    case price.NAME: {
+      return await price.execute(request, config)
+    }
+    case dominance.NAME: {
+      return await dominance.execute(request, config)
+    }
+    case marketcap.NAME: {
+      return await marketcap.execute(request, config)
+    }
+    default: {
+      throw new AdapterError({
+        jobRunID,
+        message: `Endpoint ${endpoint} not supported.`,
+        statusCode: 400,
+      })
+    }
+  }
+}
+
+export const makeExecute: ExecuteFactory = (config) => {
```

Comment:
Should be updated to the latest change, in other places as well.
```suggestion
export const makeExecute: ExecuteFactory<Config> = (config) => {
```